src/messaging/ExchangeMgr.cpp

```diff
@@ -203,23 +236,43 @@ void ExchangeManager::DispatchMessage(const PacketHeader & packetHeader, const P
             }
         }
     }
-    // Discard the message if it isn't marked as being sent by an initiator.
-    else
+    // Discard the message if it isn't marked as being sent by an initiator and the message does not needs to send
+    // ack to the peer.
+    else if (!payloadHeader.IsNeedsAck())
     {
         ExitNow(err = CHIP_ERROR_UNSOLICITED_MSG_NO_ORIGINATOR);
     }
 
+    // Create new exchange to send ack for a duplicate message and then close this exchange.
```

Comment:
If we didn't find an existing exchange that matches the message, and no unsolicited message handler registered
to hand this message, it must be a duplicated message supposing all nodes support the same set of protocols. It might not true if peer node comply different version of spec or register different set of unsolicited message handler.

Update the comments, we need to create a temporary exchange to send ack since all messages need to be acked as long as the needsAck flag is set.

Reply:
> supposing all nodes support the same set of protocols

Right, if everyone is behaving.  Which is not at all guaranteed, but is _one_ way we could land here.

> since all messages need to be acked as long as the needsAck flag is set.

Again, that's not what the CHIP spec says, but I assume that will be sorted out in the followup PR.  In the meantime, a TODO comment about it needing to be sorted out would be a good idea.

Reply:
This will be checked https://github.com/project-chip/connectedhomeip/issues/4202